pkg/controllers/repositoryserver/handler.go

```diff
@@ -284,7 +287,7 @@ func (h *RepoServerHandler) preparePodOverride(ctx context.Context) (map[string]
 	return podOverride, nil
 }
 
-func (h *RepoServerHandler) updateServerInfoInCRStatus(ctx context.Context, info crv1alpha1.ServerInfo) error {
+func (h *RepoServerHandler) updateServerInfoInCRStatus(ctx context.Context, podName string, serviceName string) error {
```

Comment:
I see that this method is called when we create Pod as well as when we create service. When we call it after creating pod there wouldnt be a service for it and even after that we are always setting service name to passed service name?

Reply:
> Could you please point me to where are we directly referencing service?
I am not sure what do you mean by this. I didn't say that we are directly referencing service.

> I think first we are creating the service and then creating the pod.
Can you check if that's what is happening.

Reply:
And if we are creating the service first, what would be the value of pod when we call `updateServerInfoInCRStatus(ctx context.Context, podName string, serviceName string)`?

Reply:
so in the first run, the pod value would be blank, and lets say the further steps failed, then on further cycle, the pod value would be set as in the previous one.

Reply:
my concern is if we know that podname is empty why do we still pass it in the function?

Reply:
@viveksinghggits The values that are passed to this function when
1. After `create service` is successful,
  ```
   h.updateServerInfoInCRStatus(ctx, h.RepositoryServer.Status.ServerInfo.PodName, svc.Name)
  ```
This means we are setting the the pod name to whatever the field value we have in repository server CR currently. So we are rewriting the pod name with its exisitng value from the repository server CR's status. Hence while creating service only service name is updated with new value

2. After `create Pod` is successful

``` h.updateServerInfoInCRStatus(ctx, pod.Name, h.RepositoryServer.Status.ServerInfo.ServiceName)``` . Here we are rewriting the service name to its existing value from repository server CR's status. While creating pod, only pod name is updated with new value

Does this address your concern?

Reply:
mm, I think so.
Maybe this question should have been part of other PR because the change doesn't seem to be related to this PR directly. I will not block the PR anymore and approve.

Reply:
Thanks @kale-amruta 👍🏻